core/camel-xml-io/src/test/java/org/apache/camel/xml/out/ModelWriterTest.java

```diff
@@ -0,0 +1,219 @@
+                        .cause(e)
+                        .message(message) //
+                        .expected(expected) //
+                        .actual(actual) //
+                        .buildAndThrow();
+            }
+        } catch (IllegalAccessException e) {
+            assertionFailure().cause(e).buildAndThrow();
+        }
+    }
+
+    private static void deepEquals(Object expected, Object actual, String path) throws IllegalAccessException {
+        if (expected == null || actual == null) {
+            assertSame(expected, actual, path);
+        } else if (expected.getClass() != actual.getClass()) {
+            fail("Not equals at " + path);
+        } else if (expected instanceof Collection) {
+            Iterator<?> ie = ((Collection) expected).iterator();
+            Iterator<?> ia = ((Collection) actual).iterator();
+            int i = 0;
+            while (ie.hasNext() && ia.hasNext()) {
+                deepEquals(ie.next(), ia.next(), path + "[" + (i++) + "]");
+            }
+            assertEquals(ie.hasNext(), ia.hasNext(), path);
+        } else if (expected.getClass() == AtomicBoolean.class) {
+            assertEquals(((AtomicBoolean) expected).get(), ((AtomicBoolean) actual).get(), path);
+        } else if (expected.getClass().isEnum()) {
+            assertEquals(((Enum) expected).name(), ((Enum) actual).name(), path);
+        } else if (expected.getClass().getName().startsWith("java.")) {
+            assertEquals(expected, actual, path);
+        } else {
+            for (Class<?> clazz = expected.getClass(); clazz != Object.class; clazz = clazz.getSuperclass()) {
+                for (Field field : clazz.getDeclaredFields()) {
+                    if (isTransient(field)) {
+                        continue;
+                    }
+                    field.setAccessible(true);
+                    Object fe = field.get(expected);
+                    Object fa = field.get(actual);
+                    deepEquals(fe, fa, (path.length() > 0 ? path + "." : path) + field.getName());
+                }
+            }
+        }
+    }
+
+    private static Map<Field, Boolean> TRANSIENT = new ConcurrentHashMap<>();
+
+    private static boolean isTransient(Field field) {
+        return TRANSIENT.computeIfAbsent(field, ModelWriterTest::checkIsTransient);
+    }
+
+    private static boolean checkIsTransient(Field field) {
+        if (Modifier.isStatic(field.getModifiers())) {
+            return true;
+        }
+        if (field.getAnnotation(XmlTransient.class) != null) {
+            return true;
+        }
+        String name = field.getName();
+        try {
+            Method method = field.getDeclaringClass().getDeclaredMethod(
+                    "get" + name.substring(0, 1).toUpperCase() + name.substring(1));
+            if (method.getAnnotation(XmlTransient.class) != null) {
+                return true;
+            }
+        } catch (Throwable t) {
+            // ignore
+        }
```

Comment:
We should not catch `Throwable`. This would also catch JVM errors (such as OOME) and could hide some serious problems. In general, catching just Exception should be fine.